### PR TITLE
fix(renderer): fix tdf renderer bug

### PR DESCRIPTION
**Pull request by etkmao**
# Pre-PR Checklist

- [x] I added/updated relevant documentation.
- [x] I followed the [Convention Commit](https://conventionalcommits.org/) guideline with maximum 72 characters to submit commit message.
- [x] I squashed the repeated code commits.
- [x] I signed the [CLA].
- [x] I added/updated test cases to check the change I am making.
- [x] All existing and new tests are passing.

### CODE REVIEW

File: modules/footstone/src/worker.cc [hunk 1]
```diff
@@ -61,6 +61,7 @@ Worker::Worker(std::string name, bool is_schedulable, std::unique_ptr<Driver> dr
 
 Worker::~Worker() {
   FOOTSTONE_CHECK(driver_->IsTerminated()) << "Terminate function must be called before destruction";
+  thread_.detach();
```
Comment (churchill-zhang) [inline modules/footstone/src/worker.cc:64]: worker 的 Terminate 有 join，这里不需要 detach

File: modules/footstone/src/hippy_value.cc [hunk 2]
```diff
@@ -376,9 +376,16 @@ bool HippyValue::IsUInt32() const noexcept { return type_ == Type::kNumber && nu
...
+    i32 = num_.i32_;
+    return true;
+  } else if (IsDouble()) {
+    i32 = static_cast<int32_t>(num_.d_);
+    if (i32 == num_.d_) {
+      return true;
+    }
+  }
+  return false;
 }
```
Comment (ilikethese) [inline modules/footstone/src/hippy_value.cc:389]: 为什么这里要把 double 转成int ？

**Review by churchill-zhang — APPROVED** [review]

**Review by ilikethese — APPROVED** [review]

**Review by vimerzhao — DISMISSED** [review]

**Review by siguangli — APPROVED** [review]
